packages/http/src/Adaptor.js

```diff
@@ -43,87 +35,69 @@ export function execute(...operations) {
   };
 }
 
-var cookiejar = new tough.CookieJar();
-var Cookie = tough.Cookie;
+function handleResponse(state, response) {
+  return {
+    ...composeNextState(state, response.body),
+    response,
+  };
+}
 
-axios.interceptors.request.use(config => {
-  cookiejar?.getCookies(config.url, (err, cookies) => {
-    config.headers.cookie = cookies?.join('; ');
-  });
-  return config;
-});
+function handleCallback(state, callback) {
+  if (callback) return callback(state);
+  return state;
+}
 
-function handleCookies(response) {
-  const { config, data, headers } = response;
-  if (config.keepCookie) {
-    let cookies;
-    let keepCookies = [];
+function request(method, path, params, callback) {
+  return state => {
+    const [resolvedPath, resolvedParams] = expandReferences(
+      state,
+      path,
+      params
+    );
 
-    if (headers['set-cookie']) {
-      if (headers['set-cookie'] instanceof Array) {
-        cookies = headers['set-cookie']?.map(Cookie.parse);
-      } else {
-        cookies = [Cookie.parse(headers['set-cookie'])];
-      }
+    let body = resolvedParams?.body;
+    let headers = resolvedParams?.headers;
 
-      headers['set-cookie']?.forEach(c => {
-        cookiejar.setCookie(Cookie.parse(c), config.url, (err, cookie) => {
-          keepCookies?.push(cookie?.cookieString());
-        });
+    if (resolvedParams?.form) {
+      let form = new FormData();
+      Object.entries(resolvedParams?.form).forEach(element => {
+        form.append(element[0], element[1]);
       });
-    }
-
-    const extendableData = Array.isArray(data) ? { body: data } : data;
 
-    return {
-      ...response,
-      data: {
-        ...extendableData,
-        __cookie: keepCookies?.length === 1 ? keepCookies[0] : keepCookies,
-        __headers: response.headers,
-      },
-    };
-  }
+      body = form;
+    } else {
+      body = body ? JSON.stringify(body) : null;
+    }
 
-  return response;
-}
+    const baseUrl = state.configuration?.baseUrl;
 
-function handleResponse(state, response) {
-  console.log(
-    response.config.method.toUpperCase(),
-    'request succeeded with',
-    response.status,
-    '✓'
-  );
+    const auth = addBasicAuth(state.configuration, headers || {});
 
-  const compatibleResp = {
-    ...response,
-    httpStatus: response.status,
-    message: response.statusText,
-    data: tryJson(response.data),
-  };
+    const maxRedirections =
+      resolvedParams?.maxRedirections ??
+      (resolvedParams?.followAllRedirects === false ? 0 : 5);
 
-  const respWithCookies = handleCookies(compatibleResp);
+    const options = {
+      ...resolvedParams,
+      ...auth,
+      baseUrl,
+      body,
+      maxRedirections,
+    };
 
-  return {
-    ...composeNextState(state, respWithCookies.data),
-    response: respWithCookies,
+    return commonRequest(method, resolvedPath, options)
+      .then(response => handleResponse(state, response))
```

Comment:
Both handleResponse and handleCallback are so lightweight... it's almost MORE confusing to have the extra `.then()` steps. Unless they're re-used elsewhere,should we not just do it all here?